plugins/experimental/client_allow_list/unit_tests/test.cc

```diff
@@ -0,0 +1,360 @@
+
+  reset_globals();
+  {
+    char const *args[] = {"dummy plugin name", DIR "bad2.yaml"};
+    REQUIRE(bad_config(2, args));
+  }
+  UT_CHECK_OUTPUT("client_allow_list: YAML::Exception \"invalid node; this may result from using a map"
+                  " iterator as a sequence iterator, or vice-versa\" when parsing YAML config file ./" DIR "bad2.yaml\n");
+
+  reset_globals();
+  {
+    char const *args[] = {"dummy plugin name", DIR "bad3.yaml"};
+    REQUIRE(bad_config(2, args));
+  }
+  UT_CHECK_OUTPUT("client_allow_list: empty server name list\n"
+                  "client_allow_list: config error: file=./" DIR "bad3.yaml line=2 column=1\n");
+
+  reset_globals();
+  {
+    char const *args[] = {"dummy plugin name", DIR "bad4.yaml"};
+    REQUIRE(bad_config(2, args));
+  }
+  UT_CHECK_OUTPUT("client_allow_list: blank space not allowed in server name list\n"
+                  "client_allow_list: config error: file=./" DIR "bad4.yaml line=2 column=1\n");
+
+  reset_globals();
+  {
+    char const *args[] = {"dummy plugin name", DIR "bad5.yaml"};
+    REQUIRE(bad_config(2, args));
+  }
+  UT_CHECK_OUTPUT("client_allow_list: <none> used more than once\n"
+                  "client_allow_list: config error: file=./" DIR "bad5.yaml line=4 column=1\n");
+
+  reset_globals();
+  {
+    char const *args[] = {"dummy plugin name", DIR "bad6.yaml"};
+    REQUIRE(bad_config(2, args));
+  }
+  UT_CHECK_OUTPUT("client_allow_list: <other> used more than once\n"
+                  "client_allow_list: config error: file=./" DIR "bad6.yaml line=4 column=1\n");
+
+  reset_globals();
+  {
+    char const *args[] = {"dummy plugin name", DIR "bad7.yaml"};
+    REQUIRE(bad_config(2, args));
+  }
+  UT_CHECK_OUTPUT("client_allow_list: cert names for SNI server name \"twice\" previously specified\n"
+                  "client_allow_list: config error: file=./" DIR "bad7.yaml line=4 column=1\n");
+
+  reset_globals();
+  {
+    char const *args[] = {"dummy plugin name", DIR "bad8.yaml"};
+    REQUIRE(bad_config(2, args));
+  }
+  UT_CHECK_OUTPUT("client_allow_list: YAML::Exception \"yaml-cpp: error at line 4, column 3: end of map"
+                  " not found\" when parsing YAML config file ./" DIR "bad8.yaml\n");
+
+  reset_globals();
+  {
+    char const *args[] = {"dummy plugin name", DIR "bad9.yaml"};
+    REQUIRE(bad_config(2, args));
+  }
+  UT_CHECK_OUTPUT("client_allow_list: YAML::Exception \"yaml-cpp: error at line 3, column 3: bad conversion\""
+                  " when parsing YAML config file ./" DIR "bad9.yaml\n");
+
+  reset_globals();
+  {
+    char const *args[] = {"dummy plugin name", DIR "bad10.yaml"};
+    REQUIRE(bad_config(2, args));
+  }
+  UT_CHECK_OUTPUT("client_allow_list: YAML::Exception \"yaml-cpp: error at line 4, column 5: the referenced "
+                  "anchor is not defined\" when parsing YAML config file ./" DIR "bad10.yaml\n");
+
+  return 0;
+}
+
+#include <../src/tscore/HashFNV.cc>
```

Comment:
Do we need this? Seems like ssl_session_reuse uses HashFNV without including the cc files. And there are some plugins that include files under include/tscore.

Reply:
These includes are only for unit test / make check.  Is there a better way to access this code in the unit test executables.

Reply:
Tests for access_control and sslheaders link libtscore. I'm not sure if it's better but looks more normal.

Reply:
Yes that would work, but it makes the dependencies of the test executable more ambiguous.